### PR TITLE
Read more than the first sector of the GPS log (#28)

### PR DESCRIPTION
**Pull request by mbof**
The raw GPS log can contain multiple 4kB sectors.
Before this change, only the first such sector was read.
After this change, all the sectors are read.
Ran tests with pytest.
Tested against HX590: gpslog -g used to return 201 points, now returns 726 points. (4 sectors are present in the raw file.)
Tested gpslog -j and gpslog -p as well.

### CODE REVIEW

File: hxtool/cli/gpslog.py [hunk 1]
```diff
@@ -141,8 +141,8 @@ def write_json(log_data: bytes, file_name: str) -> int:
         logger.warning("Log is blank. Not writing empty JSON log")
         return 0
     jlog = {
-        "header": hexlify(bytes(log.header)).decode("ascii"),
-        "unknown": hexlify(log.unknown_3c).decode("ascii"),
+        "header": hexlify(bytes(log.sectors[0].header)).decode("ascii"),
+        "unknown": hexlify(log.sectors[0].unknown_3c).decode("ascii"),
```
Comment (mbof) [inline hxtool/cli/gpslog.py:145]: I wasn't sure why the raw log header bytes were included in the JSON output, so I left the header of the first sector to not change this part of the output. If something more appropriate is warranted, pls let me know.

**Comment by cr** [comment]
Oh, that's a sector ID, brilliant! You just advanced our understanding of the binary format.

Thanks for adding test data and code. Are you sufficiently certain that the binary data doesn't contain any waypoints or other data you don't want published on the Internet?

File: hxtool/cli/gpslog.py (hunk 1 above)
Comment (cr) [reply hxtool/cli/gpslog.py:145]: It's been a while, but if I remember correctly, the intention of including all the data was to get bijective conversion between the two formats.

Comment (mbof) [reply hxtool/cli/gpslog.py:145]: OK. I'm not sure how to do this with logs having more than one sector — the sector header information seems to be specific to the raw layout... I could add all the sector headers here but I'm not sure it'd be very helpful. Is what the code is doing here OK or should it do something else? Thanks.

**Comment by mbof** [comment]
Thanks for checking re. the binary test data. It is from a recording I made recently and is OK to include.

Re. the format, I found this doc on Locus mentioning the sector layout on page 12: https://simcomturkiye.com/pdf/GNSS/SIM28/Locus_Manual_for_MTK_GNSS_Platform_V1.00.pdf (mentioning this

**Comment by cr** [comment]
The old dump format doesn't reflect a world where the blocks are chained, so its whole concept needs to be re-visited. For now I think it's more important to fix the log truncation than worrying about the dump format, so let's just merge and figure out the rest from there. Big thanks for the patch!

If you find the time to wrap your head around the format considerations, please feel free to give it a try. I likely won't be getting to it in the next few weeks.

**Pull request merged**